src/kibana/components/courier/data_source/_abstract.js

```diff
@@ -295,6 +295,23 @@ define(function (require) {
           };
 
           /**
+           * Evict Filters that doesn't match the index patterns
+           * @param {object} filter The filter to clean
+           * @returns {object}
+           */
+          var keepApplicableFilters = function (filter) {
+            if (!!(filter.index)) {
+              return filter.index === this.id;
+            } else if (!!(filter.meta) && !!(filter.meta.index)) {
+              return filter.meta.index === this.id;
+            } else if (!!(filter.geo_bounding_box)) {
+              var field = _.keys(filter.geo_bounding_box)[0];
```

Comment:
We don't look for any particular type of filter in this function otherwise. Why is this required? Seems like if this is needed some level of abstraction should be added to allow for other filters that might need this functionality.